h3.go

```diff
@@ -67,6 +67,59 @@ const (
 	RadsToDegs = 180.0 / math.Pi
 )
 
+var H3ErrorFailed = errors.New("Unknown error")
+var H3ErrorDomain = errors.New("Argument out of range error")
+var H3ErrorLatLngDomain = errors.New("Latitude or longitude out of range error")
+var H3ErrorResDomain = errors.New("Resolution out of range error")
+var H3ErrorInvalidIndex = errors.New("H3Index invalid error")
+var H3ErrorInvalidDirEdge = errors.New("Directed edge invalid error")
+var H3ErrorInvalidUnDirEdge = errors.New("Undirected edge invalid error")
+var H3ErrorInvalidVertex = errors.New("Vertex invalid error")
+var H3ErrorPentagon = errors.New("Pentagon distortion encountered error")
+var H3ErrorDupInput = errors.New("Duplicate input error")
+var H3ErrorNeighbors = errors.New("Cells are not neighbors error")
+var H3ErrorResMismatch = errors.New("Resolution mismatch error")
+var H3ErrorMemAlloc = errors.New("Memory allocation error")
+var H3ErrorMemBound = errors.New("Memory bounds error")
+var H3ErrorInvalidOption = errors.New("Invalid mode or flag error")
+
+func New(errc int) error {
```

Comment:
```suggestion
func errFromCode(errc int) error {
```

suggest unexport the error mapper as it is not relevant to consumers and rename to reflect that it maps from h3 error codes from C to Go